feat(docs): add api-extractor config

iankhou opened the pull request:
Allows us to invoke `api-extractor` to get an outputted JSON file produced from source code and TSDocs.

Ref: https://api-extractor.com/pages/setup/invoking/#4-running-the-tool

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license

iankhou committed:
add api-extractor config

rix0rrr reviewed (commented):
How/where are you planning to run API Extractor? I'd expect the script and the config to live side-by-side (unless there's a very good reason not to).

Also, if it must live here: I think I'd prefer all of our JSON files that are actually config to go through projen. So create a new [projen.JsonFile](https://projen.io/docs/api/projen/#jsonfile-) in `.projenrc.ts` and put the config there I think.

tacck committed:
chore: Fix URLs in md (#297)

Fixes #296 

This change is to fix URLs that return 404 errors.

---
By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache-2.0 license

Tietew committed:
feat(cli): CcApi context provider can be configured to fail if listing does not find a specific count of resources (#251)

Fixes #257

As described in the issue, `CcApiContextProviderPlugin` should have the
way to ensure the result has exact one resource.

This PR adds the `expectedMatchCount` option to restrict the length of
results in `listResources()`.

Unit tests are added and modified ensure the changes.

---
By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache-2.0 license

---------

Co-authored-by: Rico Huijbers <rix0rrr@gmail.com>

iankhou committed:
generate api-extractor config using projen

api-extractor doc generation and upload
